spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/repository/query/PartTreeDatastoreQuery.java

```diff
@@ -182,7 +182,7 @@ private Object execute(Object[] parameters, Class returnedElementType, Class<?>
 				|| (this.tree.isDelete() && returnedTypeIsNumber) || total;
 
 		Collector<?, ?, ?> collector = Collectors.toList();
-		if (isCountingQuery) {
+		if (isCountingQuery && !this.tree.isDelete()) {
```

Comment:
could you clarify why did you change that class?

Reply:
It turns out we had a situation that was not covered by a test. 

If the query method was a delete method but required the return of the count of deleted IDs, then we would use the Collectors.counting() collector to iterate through the raw result and then we pass that raw result to the delete function. however, the iterable had already been used once and is at its end , so the delete operation would delete no IDs from the iterable.